docs/sources/shared/flow/reference/components/http-client-proxy-config-description.md

```diff
@@ -0,0 +1,19 @@
+---
+aliases:
+- /docs/agent/shared/flow/reference/components/http-client-proxy-config-description-args/
+- /docs/grafana-cloud/agent/shared/flow/reference/components/http-client-proxy-config-description-args/
+- /docs/grafana-cloud/monitor-infrastructure/agent/shared/flow/reference/components/http-client-proxy-config-description-args/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/shared/flow/reference/components/http-client-proxy-config-description-args/
+- /docs/grafana-cloud/send-data/agent/shared/flow/reference/components/http-client-proxy-config-description-args/
+canonical: https://grafana.com/docs/agent/latest/shared/flow/reference/components/http-client-proxy-config-description-args/
+description: Shared content, http client config description
+headless: true
+---
+
+`no_proxy` can contain IPs, CIDR notation and domain names. IP and domain names can contain port numbers.
+If `no_proxy` is configured, `proxy_url` must be configured.
+
+`proxy_from_environment` uses the environment variables HTTP_PROXY, https_proxy, HTTPs_PROXY, https_proxy, and no_proxy.
```

Comment:
This is throwing me off a bit, especially with the mixed casing. We might want to copy Go's phrasing here:

> ProxyFromEnvironment returns the URL of the proxy to use for a given request, as indicated by the environment variables HTTP_PROXY, HTTPS_PROXY and NO_PROXY (or the lowercase versions thereof). Requests use the proxy from the environment variable matching their scheme, unless excluded by NO_PROXY.